chapter/related-work.md

```diff
@@ -6,6 +6,90 @@ As such, it is important to get a picture of the field of current LSP projects.
 This chapter will survey a varied range of popular language servers, compare common capabilities, and implementation approaches.
 Additionally, this part aims to recognize alternative approaches to the LSP, in the form of legacy protocols, extensible development platforms LSP extensions and the emerging Language Server Index Format.
 
+
+## Previous Approaches
+
+### IDEs
+
+Before the invention of the Language Server Protocol, extensive language support used to be provided by an IDE.
+Yet, the range of officially supported languages remained relatively small [@intellij-supported-languages].
+While integration for popular languages was common, IDE grade support for less popular ones was all but guaranteed and relied mainly on community efforts.
+In fact Eclipse[@eclipse-a-platform,eclipse-www], IntelliJ[@intelliJ], and Visual Studio[@VisualStudio], to this day the most popular IDE choices, Focus on a narrow subset of languages, historically, Java and .NET.
+Additional languages can be integrated by custom (third-party) plugins or derivations of the base platform ([@list-of-eclipse,@jetbrains-all-products]).
```

Comment:
```suggestion
Additional languages can be integrated by custom (third-party) plugins or extensions of the base platform ([@list-of-eclipse,@jetbrains-all-products]).
```
I don't know, but derivations sounded a bit strange in this context. Maybe it's Nix influence

Reply:
With derivations i mean all the different Eclipse based ide's.
Although they are in the end just plugins too

Reply:
Oh, then I think maybe the right word is "derivative" instead? It does sound more familiar than derivation, which I think rather refers to the act, the process of making a derivative (the product).